src/components/external-source/ExternalSourcesPanel.svelte

```diff
@@ -0,0 +1,189 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import {
+    derivationGroupPlanLinkError,
+    derivationGroups,
+    derivationGroupsLastAcknowledged,
+    externalSources,
+    planDerivationGroupLinks,
+  } from '../../stores/external-source';
+  import { plan } from '../../stores/plan';
+  import type { User } from '../../types/app';
+  import type { DerivationGroup, ExternalSourceSlim } from '../../types/external-source';
+  import type { ViewGridSection } from '../../types/view';
+  import effects from '../../utilities/effects';
+  import { tooltip } from '../../utilities/tooltip';
+  import Collapse from '../Collapse.svelte';
+  import CollapsibleListControls from '../CollapsibleListControls.svelte';
+  import GridMenu from '../menus/GridMenu.svelte';
+  import AlertError from '../ui/AlertError.svelte';
+  import CardList from '../ui/CardList.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import ExternalSourcePanelEntry from './ExternalSourcePanelEntry.svelte';
+  import ExternalSourceUpdateCard from './ExternalSourceUpdateCard.svelte';
+
+  export let gridSection: ViewGridSection;
+  export let user: User | null;
+
+  // filter which derivation groups are visible
+  let filterText: string = '';
+  let mappedDerivationGroups: { [key: string]: DerivationGroup[] } = {};
+  let unseenSources: ExternalSourceSlim[] = [];
+  let filteredDerivationGroups: DerivationGroup[] = [];
+
+  $: if ($plan !== null && $plan.id !== null) {
+    console.log($derivationGroupsLastAcknowledged[$plan.id]);
+  }
+
+  // Determine which new and deleted sources are unacknowledged for the user
+  $: {
+    if ($plan !== null && $plan.id !== null && $derivationGroupsLastAcknowledged[$plan.id] !== undefined) {
+      // a mapping of derivation groups (that are associated with this plan) to the time their last update was last acknowledged
+      const planDerivationGroupLastAcknowledgedAt = $derivationGroupsLastAcknowledged[$plan.id];
+      const groups = Object.keys(planDerivationGroupLastAcknowledgedAt).filter(
+        group => !planDerivationGroupLastAcknowledgedAt[group].acknowledged,
+      );
+      unseenSources = $externalSources.filter(externalSource => {
+        if (!groups.includes(externalSource.derivation_group_name)) {
+          return false;
+        }
+        // it was acknowledged recently...so skip
+        if (planDerivationGroupLastAcknowledgedAt[externalSource.derivation_group_name].acknowledged) {
+          return false;
+        }
+        // check dates
+        return (
```

Comment:
If the `acknowledged` boolean is false, can we be confident that that means it should be shown without having to check the timestamps?

Reply:
Yep! It defaults to true, and is only ever set to false on uploads of external sources (the database sets it to false then). Nobody else should or currently does set it to false ever. The UI only ever sets it back from false to true in an effect.

Reply:
No, we do still need a date check. `acknowledged` tells us that there is stuff that hasn't been acknowledged, but as for which `m` of the `n` sources in a group are unacknowledged, we need the date comparison!

Reply:
Gooot it! Okay thanks!